api/services_cluster.go

```diff
@@ -0,0 +1,100 @@
+package api
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+	"github.com/gorilla/mux"
+
+	"github.com/canonical/microcloud/microcloud/api/types"
+	"github.com/canonical/microcloud/microcloud/service"
+)
+
+// ServicesClusterCmd represents the /1.0/services/cluster/{name} API on MicroCloud.
+var ServicesClusterCmd = func(sh *service.Handler) rest.Endpoint {
+	return rest.Endpoint{
+		AllowedBeforeInit: true,
+		Name:              "services/cluster/{name}",
+		Path:              "services/cluster/{name}",
+
+		Delete: rest.EndpointAction{Handler: authHandler(sh, removeClusterMember), AllowUntrusted: true},
+	}
+}
+
+// removeClusterMember removes the given cluster member from all services that it exists in.
+func removeClusterMember(state *state.State, r *http.Request) response.Response {
+	force := r.URL.Query().Get("force") == "1"
+	name, err := url.PathUnescape(mux.Vars(r)["name"])
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	supportedServices := map[types.ServiceType]string{
+		types.MicroOVN:  MicroOVNDir,
+		types.MicroCeph: MicroCephDir,
+		types.LXD:       LXDDir,
+	}
+
+	existingServices := []types.ServiceType{types.MicroCloud}
+	for serviceType, stateDir := range supportedServices {
+		if service.Exists(serviceType, stateDir) {
+			existingServices = append(existingServices, serviceType)
+		}
+	}
+
+	addr, _, err := net.SplitHostPort(state.Address().URL.Host)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("State address %v is invalid: %w", state.Address().String(), err))
```

Comment:
`%q` instead of `%v`?

Reply:
You can use `reponse.InternalError()` here.

Reply:
`SmartError` falls back to `InternalError`.

Reply:
True, I was just thinking that makes it more readable as someone who doesn't know what `SmartError` is doing knows what is effectively being returned here. 

But fine for me as this pattern is also heavily used at other places.